### ISSUE TITLE
Bad Assertion in ResultCacheState. UnusedLocalVariable rule with null `metric` prop

### ISSUE THREAD

**Issue by Floberrot** [phpmd/phpmd#1279]
- **PHPMD version:** 3.x.dev
- **PHP Version:** 8.4.18
- **Installation type:** composer
- **Operating System / Distribution & Version:** Linux

### Current Behavior
When using PHPMD 3.x with the caching feature enabled (`--cache` and `--cache-file`), an `AssertionError` is triggered during the analysis of certain files. Specifically, when a rule like `PHPMD\Rule\UnusedLocalVariable` is triggered, the `metric` property of the violation is `null`. 

The `ResultCacheState::getRuleViolations` method contains an assertion that expects the metric to be numeric, which fails when it is `null`, causing the process to crash.

**Error Output:**
```bash
phpmd: @php vendor/bin/phpmd analyze app config database features routes --format ansi --ruleset phpmd.xml --cache --cache-file=storage/framework/cache/phpmd.cache.php
^ array:10 [
  "rule" => "PHPMD\Rule\UnusedLocalVariable"
  "namespaceName" => null
  "className" => null
  "methodName" => null
  "functionName" => null
  "beginLine" => 200
  "endLine" => 200
  "description" => "Avoid unused local variables such as '$email'."
  "args" => array:1 [
    0 => "$email"
  ]
  "metric" => null
]
[05-Mar-2026 10:28:11 UTC] PHP Fatal error:  Uncaught AssertionError: assert(is_numeric($violation['metric'])) in /var/www/html/vendor/phpmd/phpmd/src/Cache/Model/ResultCacheState.php:132
Stack trace:
#0 /var/www/html/vendor/phpmd/phpmd/src/Cache/Model/ResultCacheState.php(132): assert(false, 'assert(is_numer...')
#1 /var/www/html/vendor/phpmd/phpmd/src/Cache/ResultCacheUpdater.php(32): PHPMD\Cache\Model\ResultCacheState->getRuleViolations('/var/www/html', Array)
...
```

### Expected Behavior
PHPMD should complete the analysis successfully without crashing when the cache is enabled, even if some rules return a `null` metric. The assertion should account for the fact that a violation metric can be nullable.

### Steps To Reproduce
1.  Introduce a violation that triggers the `UnusedLocalVariable` rule.
2.  Run PHPMD with the `--cache` and `--cache-file` flags one first time (without cache).
3.  Rerun the command a second time.
4.  The analysis will crash with an `AssertionError` in `ResultCacheState.php`.

### Suggested Fix
The assertion in `src/Cache/Model/ResultCacheState.php` on line 131 should be updated to allow `null` values, as many rules (like `UnusedLocalVariable`) do not provide a numeric metric.

**Proposed change:**
```php
// src/Cache/Model/ResultCacheState.php:131
assert($violation['metric'] === null || is_numeric($violation['metric']));
```

**Comment by Floberrot** [comment]
(the error throw again with `UnusedFormalParameter` rule btw, i do not know if it helps you)
